pages/learn/[slug].tsx

```diff
@@ -0,0 +1,59 @@
+import { useRouter } from "next/router"
+import { createPage } from "../../components/page"
+import {
+  Basics,
+  Role,
+  Write
+} from "../../components/LearnTestimonyComponents/LearnComponents"
+import Tabs from "../../components/Tabs/Tabs"
+
+type TabsType = {
+  label: string
+  slug: string
+  index: number
+  Component: React.FC<{}>
+}
+
+const tabs: TabsType[] = [
+  {
+    label: "The Basics of Testimony",
+    slug: "basics-of-testimony",
+    index: 1,
+    Component: Basics
+  },
+  {
+    label: "The Role of Testimony",
+    slug: "role-of-testimony",
+    index: 2,
+    Component: Role
+  },
+  {
+    label: "Writing Effective Testimony",
+    slug: "writing-effective-testimony",
+    index: 3,
+    Component: Write
+  }
+]
+
+export default createPage({
```

Comment:
This is *awesome*. Using a path is much cleaner than a `tab` query parameter. As it is, client-side routing to these pages works but not direct navigation (refreshing or entering the url from a new tab). We need to tell Next.js which pages to generate during the build. I was able to get things working by adding [getStaticPaths and getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths) to the page:

```ts
export const getStaticPaths: GetStaticPaths = () => {
  return {
    paths: tabs.map(({ slug }) => ({ params: { slug: slug } })),
    fallback: false
  }
}

// Required by Next.js, but empty props.
export const getStaticProps: GetStaticProps = () => {
  return { props: {} }
}
```